### ISSUE TITLE
Security Problems

### ISSUE THREAD

**Issue by TheRamenChef** [gurkenlabs/litiengine#135]
`ReflectionUtilities.setValue` currently gets around Java access control by using `setAccessible`. Without a security manager, this runs fine. However, when the program is run with a security manager (with the default permissions), which it will be in typical deployment environments, it throws a `SecurityException`.

A similar problem happens in `FileUtilities.getGameResource`, when attempting to read the file on the filesystem.

**Comment by steffen-wilke** [comment]
We could provide a policy file combined with using the AccessController
https://stackoverflow.com/questions/8703234/accesscontroller-usage

This approach would require the policy file to be known when executing a LITIengine game in case the security manager is used.

Another possiblity would be to drop the support for these operation in case the security manager permits it. Using the `setAccessible` method in the reflection tool is required for reading the configuration file though...

Honestly, I'm currently not sure how to address this issue yet. Also, this particular issue didn't occur for us yet although we're running LITIengine game in various environments. Could you please specify what particular deployment environments you are talking about?

**Comment by TheRamenChef** [comment]
When I found the issue, I was running a game with the `-Djava.security.manager` option, and as far as I know, the default permissions.

When I was looking for a source to cite saying that it would be present on applications downloaded from the internet, I found I had misread the article I wanted to cite. The security manager is only present by default on web applets.

It's still important to be mindful of the security manager, though.
